src/test/regress/expected/multi_task_assignment_policy.out

```diff
@@ -268,6 +266,45 @@ SELECT count(DISTINCT value) FROM explain_outputs;
      2
 (1 row)
 
+TRUNCATE explain_outputs;
+-- test that the round robin policy detects the anchor shard correctly
+SET citus.shard_replication_factor TO 1;
+CREATE TABLE task_assignment_nonreplicated_hash (test_id  integer, ref_id integer);
+SELECT create_distributed_table('task_assignment_nonreplicated_hash', 'test_id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+BEGIN;
+SET LOCAL citus.explain_distributed_queries TO on;
+SET LOCAL citus.task_assignment_policy TO 'round-robin';
+-- run the query two times to make sure that it hits the correct worker
+-- every time
+INSERT INTO explain_outputs
+SELECT parse_explain_output($cmd$
+EXPLAIN SELECT *
+FROM (SELECT * FROM task_assignment_nonreplicated_hash WHERE test_id = 3) AS dist
+       LEFT JOIN task_assignment_reference_table ref
+                 ON dist.ref_id = ref.test_id
+$cmd$, 'task_assignment_nonreplicated_hash');
+INSERT INTO explain_outputs
+SELECT parse_explain_output($cmd$
+EXPLAIN SELECT *
+FROM (SELECT * FROM task_assignment_nonreplicated_hash WHERE test_id = 3) AS dist
+       LEFT JOIN task_assignment_reference_table ref
```

Comment:
Did you verify whether this test fails before the change? Wondering under what conditions the reference table appears first in the relation shard list.

Reply:
Yes, I verified that it fails on master branch. Checking it once again. The second `SELECT` fails with these extra outputs:
```
WARNING:  relation "public.task_assignment_nonreplicated_hash_880006" does not exist
CONTEXT:  while executing command on localhost:57637
PL/pgSQL function parse_explain_output(text,text) line 6 at FOR over EXECUTE statement
```

It should have hit the worker running on port 57638